### PR TITLE
Clean up Groovy + Add more tests

### PR DESCRIPTION
**Pull request by jaredsburrows**
@benjamin-bader 
- Cleaning up `Groovy` code
  - `void` -> `def`
  - remove semicolons
  - remove extra modifiers
  - added documentation to the extension class, more to come

- Added tests
  - added tests to the plugin class, more to come

### CODE REVIEW

File: src/main/groovy/com/getkeepsafe/dexcount/DexFile.groovy [hunk 1]
```diff
@@ -191,8 +191,8 @@ class DexFile {
      * @param zipEntries a list of ZipEntry objects inside of the APK.
      * @return a list, possibly empty, of instant-run dex data.
      */
-    private static List<DexFile> extractIncrementalDexFiles(ZipFile apk, List<ZipEntry> zipEntries) {
-        def incremental = zipEntries.findAll { it.name.equals('instant-run.zip') }
+    static List<DexFile> extractIncrementalDexFiles(ZipFile apk, List<ZipEntry> zipEntries) {
+        def incremental = zipEntries.findAll { (it.name == 'instant-run.zip') }
```
Comment (jaredsburrows) [inline src/main/groovy/com/getkeepsafe/dexcount/DexFile.groovy]: In `Groovy`, equals and `==` are the same.

File: src/main/groovy/com/getkeepsafe/dexcount/DexMethodCountExtension.groovy [hunk 2]
```diff
@@ -20,122 +20,83 @@ package com.getkeepsafe.dexcount
...
-    private boolean enableForInstantRun = false
-    private OutputFormat format = OutputFormat.LIST
-    private boolean verbose
-    private boolean printVersion
-    private int maxTreeDepth = Integer.MAX_VALUE
-    private int dxTimeoutSec = 60;
-    private String teamCitySlug = null
-    private boolean runOnEachAssemble = true
-    int maxMethodCount = -1
+    /**
```
Comment (jaredsburrows) [inline src/main/groovy/com/getkeepsafe/dexcount/DexMethodCountExtension.groovy]: Added docs to this class, more to come.

File: src/main/groovy/com/getkeepsafe/dexcount/DexMethodCountExtension.groovy [hunk 3]
```diff
@@ -20,122 +20,83 @@ package com.getkeepsafe.dexcount
...
+     */
+    boolean runOnEachAssemble = true
 
-    boolean getPrintVersion() {
-        return this.printVersion
-    }
+    /**
+     * When set, the build will fail when the APK/AAR has more methods than the max. 0 by default.
+     */
+    int maxMethodCount = -1
```
Comment (jaredsburrows) [inline src/main/groovy/com/getkeepsafe/dexcount/DexMethodCountExtension.groovy]: In `Groovy` similar to `Kotlin`, the getters/setters are auto generated.

File: src/test/groovy/com/getkeepsafe/dexcount/DexMethodCountPluginSpec.groovy [hunk 4]
```diff
@@ -0,0 +1,156 @@
+package com.getkeepsafe.dexcount
+
+import org.gradle.testfixtures.ProjectBuilder
+import spock.lang.Specification
+import spock.lang.Unroll
+
+final class DexMethodCountPluginSpec extends Specification {
```
Comment (benjamin-bader) [inline src/test/groovy/com/getkeepsafe/dexcount/DexMethodCountPluginSpec.groovy]: 🔥 🔥 🔥

**Pull request merged**